src/core/server/http/http_server.mocks.ts

```diff
@@ -18,8 +18,55 @@
  */
 import { Request, ResponseToolkit } from 'hapi';
 import { merge } from 'lodash';
+import url from 'url';
+import querystring from 'querystring';
 
-import { KibanaRequest } from './router';
+import { schema } from '@kbn/config-schema';
+
+import { KibanaRequest, RouteMethod } from './router';
+
+interface RequestFixtureOptions {
+  headers?: Record<string, string>;
+  params?: Record<string, unknown>;
+  body?: Record<string, unknown>;
+  query?: Record<string, unknown>;
+  path?: string;
+  method?: RouteMethod;
+}
+
+function createKibanaRequestMock({
+  path = '/path',
+  headers = { accept: 'something/html' },
+  params = {},
+  body = {},
+  query,
```

Comment:
**query** is passed already parsed in `KibanaRequest` and not to overcomplicate I added it as already parsed value